### ISSUE TITLE
Kernel errors with latest AMD Zen cpus (T10619)

### ISSUE THREAD

**Issue by celticmagic** [getsolus/packages#306]
<div dir="rtl"><b>TomsB</b>&nbsp;(<em>&#35;RobinHood</em>),&nbsp;2023-08-16&nbsp;14:15:05&nbsp;UTC</div><br/>

# Summary 

The last few updates (last two update syncs) brought up a strange issue for my AMD laptop. Latest kernel (-current branch) makes my laptop almost unusable. Here are my main issues:

  - laptops keyboard input is with a lag. The lag for each key press is about 1 second,
  - if pressing and holding a keyboard key for 1-2 seconds then the system thinks that the key is being presses for forever (for example Volume up key goes to 100% (and tries to go further) volume rather than being raised by 10%);
  - after some time the system decides that the Ctrl key is pressed and every action (keyboard or mouse click) is acted as Ctlr is actually pressed. For example mouse wheel scroll is suddenly page zoom in/out or mouse/toucpad click is selecting multiple items rather than just selecting the item.

## Environment

Kernel: Linux 6.4.9-248.current
Laptop: Lenovo ThinkBook 13s G4 ARB
CPU: AMD Ryzen™ 7 6800U with Radeon™ Graphics × 16

  - Is system up to date? Y 
  - Repo: Shannon  
  - Desktop Environment: Budgie 

## System details

Complete output from dmesg: {F11169216}

Here's the main error from dmesg:

```
[  106.833406] atkbd serio0: Unknown key pressed (translated set 2, code 0xf8 on isa0060/serio0).
[  106.833420] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  107.003647] atkbd serio0: Unknown key released (translated set 2, code 0xf8 on isa0060/serio0).
[  107.003657] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  108.418017] atkbd serio0: Unknown key pressed (translated set 2, code 0xf8 on isa0060/serio0).
[  108.418033] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  108.584952] atkbd serio0: Unknown key released (translated set 2, code 0xf8 on isa0060/serio0).
[  108.584966] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  115.885062] atkbd serio0: Unknown key pressed (translated set 2, code 0xf8 on isa0060/serio0).
[  115.885077] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  116.051766] atkbd serio0: Unknown key released (translated set 2, code 0xf8 on isa0060/serio0).
[  116.051783] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  281.881955] FAT-fs (nvme0n1p1): utf8 is not a recommended IO charset for FAT filesystems, filesystem will be case sensitive!
[  332.830543] mce: [Hardware Error]: Machine check events logged
[  332.830553] [Hardware Error]: Corrected error, no action required.
[  332.830563] [Hardware Error]: CPU:0 (19:44:1) MC15_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  332.830593] [Hardware Error]: Error Addr: 0x000000002032b8c0
[  332.830599] [Hardware Error]: IPID: 0x0000009600050f00, Syndrome: 0x000001ff0a240700
[  332.830609] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  332.830613] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  332.830646] mce: [Hardware Error]: Machine check events logged
[  332.830650] [Hardware Error]: Corrected error, no action required.
[  332.830905] [Hardware Error]: CPU:0 (19:44:1) MC16_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  332.830928] [Hardware Error]: Error Addr: 0x000000002030ec80
[  332.830933] [Hardware Error]: IPID: 0x0000009600150f00, Syndrome: 0x000001ff0a240701
[  332.830941] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  332.830944] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  332.830957] [Hardware Error]: Corrected error, no action required.
[  332.830961] [Hardware Error]: CPU:0 (19:44:1) MC17_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  332.830981] [Hardware Error]: Error Addr: 0x0000000020357f00
[  332.830985] [Hardware Error]: IPID: 0x0000009600250f00, Syndrome: 0x000001ff0a240701
[  332.830993] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  332.830995] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  332.831007] [Hardware Error]: Corrected error, no action required.
[  332.831011] [Hardware Error]: CPU:0 (19:44:1) MC18_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  332.831031] [Hardware Error]: Error Addr: 0x0000000020357fc0
[  332.831035] [Hardware Error]: IPID: 0x0000009600350f00, Syndrome: 0x000001ff0a240701
[  332.831043] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  332.831045] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  332.831071] [Hardware Error]: Error Addr: 0x000000002034df80
[  332.831075] [Hardware Error]: IPID: 0x0000009600350f00, Syndrome: 0x000001ff0a240700
[  332.831086] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  332.831089] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  660.511489] mce_notify_irq: 2 callbacks suppressed
[  660.511494] mce: [Hardware Error]: Machine check events logged
[  660.511499] [Hardware Error]: Corrected error, no action required.
[  660.511505] [Hardware Error]: CPU:0 (19:44:1) MC15_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  660.511525] [Hardware Error]: Error Addr: 0x000000001ead5040
[  660.511529] [Hardware Error]: IPID: 0x0000009600050f00, Syndrome: 0x000001ff0a240701
[  660.511536] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  660.511539] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  660.511554] mce: [Hardware Error]: Machine check events logged
[  660.511556] [Hardware Error]: Corrected error, no action required.
[  660.511559] [Hardware Error]: CPU:0 (19:44:1) MC16_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  660.511576] [Hardware Error]: Error Addr: 0x000000002030f440
[  660.511579] [Hardware Error]: IPID: 0x0000009600150f00, Syndrome: 0x000001ff0a240701
[  660.511585] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  660.511587] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  660.511597] [Hardware Error]: Corrected error, no action required.
[  660.511600] [Hardware Error]: CPU:0 (19:44:1) MC17_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  660.511615] [Hardware Error]: Error Addr: 0x000000002030f3c0
[  660.511618] [Hardware Error]: IPID: 0x0000009600250f00, Syndrome: 0x000001ff0a240701
[  660.511624] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  660.511626] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  660.511635] [Hardware Error]: Corrected error, no action required.
[  660.511638] [Hardware Error]: CPU:0 (19:44:1) MC18_STATUS[Over|CE|MiscV|AddrV|-|-|SyndV|CECC|-|-|-]: 0xdc204000000c011b
[  660.511654] [Hardware Error]: Error Addr: 0x000000002008af80
[  660.511657] [Hardware Error]: IPID: 0x0000009600350f00, Syndrome: 0x000001ff0a240700
[  660.511663] [Hardware Error]: Unified Memory Controller Ext. Error Code: 12
[  660.511664] [Hardware Error]: cache level: L3/GEN, tx: GEN, mem-tx: RD
[  795.497386] atkbd serio0: Unknown key pressed (translated set 2, code 0xf8 on isa0060/serio0).
[  795.497398] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
[  795.670363] atkbd serio0: Unknown key released (translated set 2, code 0xf8 on isa0060/serio0).
[  795.670377] atkbd serio0: Use 'setkeycodes e078 <keycode>' to make it known.
```

## Other comments 

Don't know if it's relevant or true to this case, but when doing online search my problem it lead me to the fact that recent Zen cpus need patches for them to work properly.
https://bbs.archlinux.org/viewtopic.php?id=285327
https://bugzilla.kernel.org/show_bug.cgi?id=216118#c10

Linux-lts (5.15.x) kernel that is found in Solus repository works without any problems and without these strange input lag and other issues.

**Comment by ReillyBrogan** [comment]
Is this still an issue?

**Comment by ReillyBrogan** [comment]
Actually, I just dug into this and this was resolved in the 6.4.11 kernel. Feel free to re-open if it's not actually fixed.
